### PR TITLE
Use order tokens in payment capture

### LINKED ISSUE #2479: Thank-you and pay URLs are accessible without login
It looks like order confirmation page does not require the user to be logged in. I just did a quick purchase on demo store and then curled the confirmation URL without cookies. Brute-forcing order token might leak personal information such as address or phone no. Here I am grepping for my fake name on the confirmation page:

```
curl https://demo3.coreshop.org/en/shop/checkout-thank-you/NoOLrFeRxd | grep -A 10 'Invoice Address'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 16072    0 16072    0     0  19993      0 --:--:-- --:--:-- --:--:-- 19990
                    Invoice Address
                </h4>
            </div>
            <div class="card-body panel-delivery-address">
                Test<br />
Mrs. Test Tester<br />
Teststr 1<br />
28219 Bremen<br />
Austria<br />

            </div>
```

"pay"-URL seem to be more dangerous. It has a form of https://demo3.coreshop.org/en/shop/pay/123. Guessing the right pay number redirects to the thank-you page of that order. The 123 seems like a incrementing integer, making brute-forcing even easier. So doing something like this can reveal personal information of all customers:

```shell
for i in $(seq 123 321); do curl -s -L https://demo3.coreshop.org/en/shop/pay/$i | grep -A 10 'Invoice Address'; done
                    Invoice Address
                </h4>
            </div>
            <div class="card-body panel-delivery-address">
                Test<br />
Mrs. Test Tester<br />
Teststr 1<br />
28219 Bremen<br />
Austria<br />

            </div>
                    Invoice Address
                </h4>
            </div>
            <div class="card-body panel-delivery-address">
                Coreshop<br />
Mrs. Corina Shopper<br />
Coreshopallee 1<br />
19999 Coreshopwille<br />
Austria<br />
0123456789
            </div>
```

I would expect this information to be accessible to the owner of the order only.

### PR DESCRIPTION
**Pull request by yariksheptykin**
Closes #2479

### CODE REVIEW

File: features/ui/frontend/checkout/prevent_repaying_completed_checkouts.feature [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    And I submit the shipping step
+    And I should be on the payment checkout step
+    And I select the payment provider "Bankwire"
+    And I submit the payment step
+    And I should be on the summary checkout step
+    And I accept the checkout terms of service
+    And I submit the order
+    And I should be on the thank you page
+
+  #
```
Comment (yariksheptykin) [inline features/ui/frontend/checkout/prevent_repaying_completed_checkouts.feature]:
```suggestion
```

**Comment by yariksheptykin** [comment]
@dpfaffenbauer I introduced a new route as you suggested and added a migration to update existing installations. Is there anything else we need to do to support BC?

**Comment by dpfaffenbauer** [comment]
@yariksheptykin Amazing, I'll give it a test and let you know. Looks good though :)

**Comment by yariksheptykin** [comment]
@dpfaffenbauer this does work at least on my local demo. Pay URL works both for token and for order id.

[Screencast from 19.01.2024 10:04:50.webm](https://github.com/coreshop/CoreShop/assets/636846/5c471409-267a-4ff2-9eb6-9dccf177c52f)

**Comment by dpfaffenbauer** [comment]
@yariksheptykin since you backported it, we will close this one and merge the one for 3.2, I will merge 3.2 into 4.0 then.

Lovely screencast :), I will still have to test it on my own. Will happen sometime next week

**Comment by yariksheptykin** [comment]
@dpfaffenbauer sounds good to me. I will patch our project (at neusta) with the backport and see it everything works in that context.

**Comment by dpfaffenbauer** [comment]
https://github.com/coreshop/CoreShop/pull/2515

**Comment by yariksheptykin** [comment]
@dpfaffenbauer https://github.com/coreshop/CoreShop/pull/2515 did not make it into v4.0.2. Is this intentional or just forgotten? Should I do anything to port the MR to v4?

**Comment by dpfaffenbauer** [comment]
@yariksheptykin are you sure? 

https://github.com/coreshop/CoreShop/blob/4.0.2/src/CoreShop/Bundle/FrontendBundle/Resources/install/pimcore/staticroutes.yml#L145
https://github.com/coreshop/CoreShop/commit/c03814b7095ea2f531ca1792c06c5647b07d4b74

**Comment by yariksheptykin** [comment]
@dpfaffenbauer ah I see you are right. I did not see this in release https://github.com/coreshop/CoreShop/releases/tag/4.0.2 and started worrying. Thanks for checking!

**Comment by dpfaffenbauer** [comment]
@yariksheptykin It is a bit confusing right. CoreShop 4.0 always gets the changes from 3.2. 3.2 changes are in that changelog, only what changed from 3.2.x to 4.0.x are in the 4.0.x changelog.

**Pull request closed without merging**